src/software/backend/simulation/physics/physics_robot.cpp

```diff
@@ -1,16 +1,55 @@
 #include "software/backend/simulation/physics/physics_robot.h"
 
-#include "shared/constants.h"
+#include <algorithm>
+#include <numeric>
+
 #include "software/backend/simulation/physics/box2d_util.h"
+#include "software/backend/simulation/physics/physics_object_user_data.h"
+extern "C"
+{
+#include "firmware/main/shared/physics.h"
+}
 
-PhysicsRobot::PhysicsRobot(std::shared_ptr<b2World> world, const Robot& robot)
+// We are only allowed to cover a fraction of the ball according to the rules, so we use
+// this to calculate the depth. We assume the ball can be dribbled as long as it is
+// anywhere within this small area.
+const double PhysicsRobot::dribbler_depth =
+    BALL_MAX_RADIUS_METERS * 2 * MAX_FRACTION_OF_BALL_COVERED_BY_ROBOT;
+// We can use a very small value for the chicker thickness since the
+// chicker just needs to be large enough to collide with the ball and detect collisions
+// without letting the ball tunnel and collide with the robot body
+const double PhysicsRobot::chicker_thickness = 0.005;
+const double PhysicsRobot::total_chicker_depth =
+    PhysicsRobot::dribbler_depth + PhysicsRobot::chicker_thickness;
+
+PhysicsRobot::PhysicsRobot(std::shared_ptr<b2World> world, const Robot& robot,
+                           double mass_kg)
     : robot_id(robot.id())
 {
-    // createRobotPhysicsBody must be called before the setup functions, so that the
-    // b2Body is instantiated before fixtures are added to it.
-    createRobotPhysicsBody(world, robot);
-    setupRobotPhysicsBody(robot);
-    setupChickerPhysicsBody(robot);
+    b2BodyDef robot_body_def;
+    robot_body_def.type = b2_dynamicBody;
+    robot_body_def.position.Set(robot.position().x(), robot.position().y());
+    robot_body_def.linearVelocity.Set(robot.velocity().x(), robot.velocity().y());
+    robot_body_def.angle           = robot.orientation().toRadians();
+    robot_body_def.angularVelocity = robot.angularVelocity().toRadians();
+    // This is a somewhat arbitrary value for damping. We keep it relatively low
+    // so that robots still coast a ways before stopping, but non-zero so that robots
+    // do come to a halt if no force is applied.
+    // Damping is roughly calculated as v_new = v * exp(damping * t)
+    // https://gamedev.stackexchange.com/questions/160047/what-does-lineardamping-mean-in-box2d
+    robot_body_def.linearDamping  = 0.2;
+    robot_body_def.angularDamping = 0.2;
```

Comment:
nit: can we put these numbers in the header file

Reply:
see above comment

Reply:
done